packages/legend-graph/src/graphManager/protocol/pure/v1/V1_PureGraphManager.ts

```diff
@@ -2281,14 +2280,23 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
   async buildHashesIndex(entities: Entity[]): Promise<Map<string, string>> {
     const hashMap = new Map<string, string>();
     const pureModelContextData = new V1_PureModelContextData();
+    const elementToEntity = new Map<V1_PackageableElement, Entity>();
     await V1_entitiesToPureModelContextData(
       entities,
       pureModelContextData,
       this.pluginManager.getPureProtocolProcessorPlugins(),
+      elementToEntity,
     );
     await Promise.all(
       pureModelContextData.elements.map((element) =>
-        promisify(() => hashMap.set(element.path, element.hashCode)),
+        promisify(() =>
+          hashMap.set(
+            elementToEntity.get(element)
```

Comment:
can you provide an explanation as to why this is needed?
Is this also something temprorary while we move over to full function signature names?